src/js/node/stream.ts

```diff
@@ -4350,7 +4354,85 @@ var require_writable = __commonJS({
     Writable.prototype[EE.captureRejectionSymbol] = function (err) {
       this.destroy(err);
     };
-    var webStreamsAdapters;
+    var webStreamsAdapters = {
+      newStreamWritableFromWritableStream(
+        writableStream: WritableStream,
+        options: {
+          highWaterMark?: number;
+          encoding?: string;
+          objectMode?: boolean;
+          signal?: AbortSignal;
+        } = {},
+      ) {
+        if (!isWritableStream(writableStream)) {
+          throw new ERR_INVALID_ARG_TYPE("writableStream", "WritableStream", writableStream);
+        }
+
+        validateObject(options, "options");
+        const {
+          highWaterMark,
+          encoding,
+          objectMode = false,
+          signal,
+          // native = true,
+        } = options;
+
+        if (encoding !== undefined && !Buffer.isEncoding(encoding)) {
+          // TODO: why does this err in typescript?
+          throw new ERR_INVALID_ARG_VALUE(encoding, "options.encoding");
+        }
+        validateBoolean(objectMode, "options.objectMode");
+
+        const nativeStream = getNativeWritableStream(writableStream, options);
+        return nativeStream;
+      },
+
+      newWritableStreamFromStreamWritable(streamWritable: any, options: any = {}) {
+        if (typeof streamWritable?._writableState !== "object") {
+          throw new ERR_INVALID_ARG_TYPE("streamReadable", "stream.Readable", streamWritable);
+        }
+        var { isDestroyed, isWritable } = require_utils();
+        if (isDestroyed(streamWritable) || !isWritable(streamWritable)) {
+          const writable = new WritableStream();
+          writable.close();
+          return writable;
+        }
+        const objectMode = streamWritable.writableObjectMode;
+        const highWaterMark = streamWritable.writableHighWaterMark;
+
+        const evaluateStrategyOrFallback = strategy => {
+          if (strategy) return strategy;
+          if (objectMode) {
+            return new CountQueuingStrategy({ highWaterMark });
+          }
+          return { highWaterMark };
+        };
+        const strategy = evaluateStrategyOrFallback(options?.strategy);
+
+        let controller;
+
+        return new WritableStream(
+          {
+            start(c) {
+              controller = c;
+            },
+
+            write(chunk) {
+              streamWritable.write(chunk);
```

Comment:
Where is backpressure signaled?

Reply:
doesn't passing in a strategy automatically signal backpressure?

Reply:
How does the Node.js Writable become aware of backpressure?